examples/launch-param-reductions.cpp

```diff
@@ -50,10 +53,25 @@ int main(int RAJA_UNUSED_ARG(argc), char** RAJA_UNUSED_ARG(argv[]))
 //
   constexpr int N = 1000000;
 
+//
```

Comment:
Thought: would the example in this file be more useful if there was a method templated on resource, exec policy, etc. that contained a generic reduction kernel and is called for each back-end in a specialized way?  That would simplify and reduce the amount of code and emphasize RAJA portability.

Reply:
I think that would be nice for all our examples. One challenge though is that resources aren't templated. We would still have to setup of the #ifdef for each type. If we wanted a generic function that contained a reduction kernel we could template on policy and pass in a typed erased the resource (still requiring the ifdefs). Then we can just call the reduction kernel function with our different backends. 

Perhaps we can discuss some ideas with a subset of the team on this.

Reply:
Sounds good. This idea  shouldn't hold up your PR. Something to work through when we have spare time.

Reply:
Take a look at what I've been trying with a `for_each_tuple` function in the multi reducer PR https://github.com/LLNL/RAJA/blob/c0b1e1b8fd4e50830bd842c19db2769d0498b21d/examples/forall_multi-reductions.cpp

Reply:
I made an old mistake in that example, I declared a host device lambda in a generic lambda which is not allowed in cuda.